packages/grid/src/vaadin-grid-data-provider-mixin.js

```diff
@@ -363,6 +364,7 @@ export const DataProviderMixin = (superClass) =>
      */
     clearCache() {
       this._dataProviderController.clearCache();
+      this._dataProviderController.rootCache.size = this.size;
```

Comment:
```suggestion
      this._dataProviderController.rootCache.size = this.size;
      this._dataProviderController.recalculateFlatSize();
      this._flatSize = this._dataProviderController.flatSize;
```

Reply:
Setting `size` on the root cache requires flat size recalculation to get `flatSize` updated. I wonder why tests didn't catch that.

Reply:
It seems it's intentional that there is no `flatSize` update here. When I add `this._flatSize = this._dataProviderController.flatSize`, the tests start freezing. They apparently expect that the grid rows will re-render only after a response is received from the data provider. I decided to keep this as it as and only added `this._dataProviderController.recalculateFlatSize()` to be on the safe side.